Source/Api/Controllers/AuthController.cs

```diff
@@ -95,33 +98,45 @@ public class AuthController : ExceptionlessApiController {
             }
 
             User user;
-            try {
-                user = await _userRepository.GetByEmailAddressAsync(model.Email);
-            } catch (Exception ex) {
-                _logger.Error().Exception(ex).Critical().Message("Login failed for \"{0}\": {1}", model.Email, ex.Message).Tag("Login").Identity(model.Email).SetActionContext(ActionContext).Write();
-                return Unauthorized();
-            }
+	        if (Settings.Current.EnableActiveDirectoryAuth && !IsAdLoginValid(model.Email, model.Password)) {
+				_logger.Error().Message("Domain login failed for \"{0}\": Invalid Password or Account.", model.Email).Tag("Login").Identity(model.Email).SetActionContext(ActionContext).Write();
+				return Unauthorized();
+	        }
+
+			try
+			{
+				user = await _userRepository.GetByEmailAddressAsync(model.Email);
+			}
+			catch (Exception ex)
+			{
+				_logger.Error().Exception(ex).Critical().Message("Login failed for \"{0}\": {1}", model.Email, ex.Message).Tag("Login").Identity(model.Email).SetActionContext(ActionContext).Write();
+				return Unauthorized();
+			}
 
             if (user == null) {
-                _logger.Error().Message("Login failed for \"{0}\": User not found.", model.Email).Tag("Login").Identity(model.Email).SetActionContext(ActionContext).Write();
-                return Unauthorized();
+				_logger.Error().Message("Login failed for \"{0}\": User not found.", model.Email).Tag("Login").Identity(model.Email).SetActionContext(ActionContext).Write();
+				return Unauthorized();
             }
 
             if (!user.IsActive) {
                 _logger.Error().Message("Login failed for \"{0}\": The user is inactive.", user.EmailAddress).Tag("Login").Identity(user.EmailAddress).Property("User", user).SetActionContext(ActionContext).Write();
                 return Unauthorized();
             }
 
-            if (String.IsNullOrEmpty(user.Salt)) {
-                _logger.Error().Message("Login failed for \"{0}\": The user has no salt defined.", user.EmailAddress).Tag("Login").Identity(user.EmailAddress).Property("User", user).SetActionContext(ActionContext).Write();
-                return Unauthorized();
-            }
+	        if (!Settings.Current.EnableActiveDirectoryAuth) {
```

Comment:
If we do the check proposed above and validate the login after checking for the null user and if they are active check, then we can just return the token and not modify any of this code.